### PR TITLE
Update grpc to 1.62.0

### LINKED ISSUE #151: latest build of abseil-cpp failed with `elaborated-type-specifier for a scoped enum must not use the ‘class’ keyword`
**Build fail**
Build fail with `elaborated-type-specifier for a scoped enum must not use the ‘class’ keyword` in 3rd party code in 
`build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:34`

Reason is a missing include of `#include <cstdint>` before.
```C++
#include <cstdint>

#include "absl/base/config.h"
#include "absl/base/port.h"
#include "absl/meta/type_traits.h"
#include "absl/strings/internal/str_format/output.h"
#include "absl/strings/string_view.h"

namespace absl {
ABSL_NAMESPACE_BEGIN

enum class FormatConversionChar : uint8_t;
enum class FormatConversionCharSet : uint64_t;
```

I was able to complete the build (a few warnings though).
I wonder whether the include statement could be done outside of the fetched as a cmake statement.

### PR DESCRIPTION
**Pull request by rainerschoe**
fixes #151 

The previously used gRPC version had build errors with more modern compilers.
Stepping to the most recent release.

### CODE REVIEW

File: third_party/gRPC_utils/cli_call.cc [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include <optional>
 // END MODIFIED
```
Comment (rainerschoe) [inline third_party/gRPC_utils/cli_call.cc:29]: I took the most recent `cli_call.cc` from gRPC example source code and adapted it again for our needs.

File: tests/functionTests/rpcExecutionTests.txt [hunk 2]
```diff
@@ -200,9 +200,7 @@ RPC succeeded :D
 
 #START_TEST json_input_stdin_malformed
 cat @@testResources@@/duplicateEverything1d_malformed.json | @@CMD@@ --disableCache --jsonInput=- 127.0.0.1 examples.NestedTypeRpcs duplicateEverything1d
-Warning: Failed to parse JSON file '-': INVALID_ARGUMENT:Unexpected end of string. Expected , or } after key:value pair.
-
-^
+/Warning: Failed to parse JSON file '-': .*
```
Comment (rainerschoe) [inline tests/functionTests/rpcExecutionTests.txt:203]: protobuf error message changed slightly. made the test more flexible

File: tests/functionTests/completionTests_fish.txt [hunk 3]
```diff
@@ -64,6 +64,7 @@ Possible Candidates:
 127.0.0.1 examples.StreamingRpcs        (uni- and bi-directional streams)
 127.0.0.1 examples.StatusHandling                   (gRPC error handling)
 /^127.0.0.1 grpc.reflection.*$
+/^127.0.0.1 grpc.reflection.*$
```
Comment (rainerschoe) [inline tests/functionTests/completionTests_fish.txt:67]: in latest version, grpc server adds two reflection services (one stable and one alpha)

File: cmake/gRPC.cmake [hunk 4]
```diff
@@ -87,7 +91,7 @@ if(GWHISPER_FORCE_BUILDING_GRPC OR GRPC_NOT_FOUND)
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "s390x")
         set(gRPC_SSL_PROVIDER "package" CACHE STRING "force overwritten by gWhisper, as boringSSL does not support s390x -> need to fall-back to system installed libssl" FORCE )
     endif()
-
+    set(ABSL_PROPAGATE_CXX_STD ON CACHE STRING "force overwritten by gWhisper to conform to Abseil recommendations" FORCE )
```
Comment (rainerschoe) [inline cmake/gRPC.cmake:94]: A build warning was given to set this. So setting here.

File: src/libCli/MessageFormatterJson.cpp [hunk 5]
```diff
@@ -27,8 +27,8 @@ namespace cli
 
         // Will print primitive fields regardless of their values. So e.g. an int32 field set to 0 will not be omitted. 
         printOptions.always_print_primitive_fields = true;
-        
-        google::protobuf::util::Status status = google::protobuf::util::MessageToJsonString(f_message, &resultString, printOptions);
+
+        auto status = google::protobuf::util::MessageToJsonString(f_message, &resultString, printOptions);
```
Comment (ja66) [inline src/libCli/MessageFormatterJson.cpp:31]: const is your friend ;-)

File: src/libCli/MessageParserJson.cpp [hunk 6]
```diff
@@ -56,7 +56,7 @@ std::vector<std::unique_ptr<google::protobuf::Message>> MessageParserJson::parse
             buffer << source->rdbuf();
             std::string fileContent = buffer.str();
             auto status = google::protobuf::util::JsonStringToMessage(
-                    google::protobuf::StringPiece(fileContent),
+                    fileContent,
```
Comment (ja66) [inline src/libCli/MessageParserJson.cpp:59]: Actually you don't need a temporay (`fileContent`) here, you could pass it directly as parameter to `JsonStringToMessage()`

**Review by ja66 — APPROVED** [review]

**Comment by rainerschoe** [comment]
Thanks Jens for reporting and reviewing :1st_place_medal: 
I assume this fixes #151

**Pull request merged**